json/parse.go

```diff
@@ -241,6 +241,10 @@ func (p *Parser) feedUntil(b []byte) (int, bool, error) {
 			return 0, false, errFailing
 		}
 
+		if err != nil {
```

Comment:
It might be better for this to just be a `break` rather than a `return` -- that way if something is consumed before the error it will still return the right partial state along with the error below.